app/templates/_exchanges.html

```diff
@@ -0,0 +1,290 @@
+            {% set status = "You need to ship" %}
+            {% set actions %}
+                <button
+                    style="margin-left: 0.25em;"
+                    onclick="updateExchange('{{ ex.id }}', 'shipped', '{{ e.id }}', 1, '{{ user.hospital.id }}') "
+                    class="btn btn-success btn-sm"
+                >Mark as Shipped</button>
+            {% endset %}
+        {% else %}
+            {% set status %}
+                Waiting for {{ e.hospital1_ref.name }} to ship
+            {% endset %}
+        {% endif %}
+    {% elif not e.is_h2_received %}
+        {% if user.hospital.id == e.hospital2 %}
+            {% set status = "Waiting for Receipt" %}
+            {% set actions %}
+                <button style="margin-left: 0.25em;"
+                onclick="updateExchange('{{ ex.id }}', 'received', '{{ e.id }}', 2, '{{ user.hospital.id }}') "
+                type="button" class="btn btn-success btn-sm">Mark as Received</button>
+            {% endset %}
+        {% else %}
+            {% set status %}
+                Waiting for {{ e.hospital2_ref.name }} to receive
+            {% endset %}
+        {% endif %}
+    {% else %}
+        {% set status = "Complete" %}
+    {% endif %}
+
+    <td>{{ status }}</td>
+    <td>{{ actions }}</td>
+{% endmacro %}
+
+{% macro exchanges_listing(title, help, exchanges) %}
+    <h3>{{ title }}</h3>
+    <table class="table table-striped">
+        <thead>
+            <tr>
+                <th scope="col">ID</th>
+                <th scope="col">Created</th>
+                <th scope="col">Updated</th>
+                <th scope="col">Exchange</th>
+                <th scope="col">Status</th>
+                <th scope="col">Actions</th>
+            </tr>
+        </thead>
+        <tbody>
+            {% if exchanges|length == 0 %}
+                <tr>
+                    <td colspan="6" style="text-align: center; font-style: italic; font-size: 1.25em;">
+                        You don't have any {{ title|lower }} exchanges right now!
+                        <br/>
+                        {{ help }}
+                    </td>
+                </tr>
+            {% endif %}
+            {% for ex in exchanges %}
+            {% for e in ex.exchange %}
+            <tr>
+                {% call whole_exchange_td(loop, ex) %}
+                    {{ ex.id }}
+                {% endcall %}
+
+                {% call whole_exchange_td(loop, ex) %}
+                    {{ ex.created_timestamp.replace(microsecond=0) }}
+                {% endcall %}
+
+                {% call whole_exchange_td(loop, ex) %}
+                    {{ ex.updated_timestamp.replace(microsecond=0) }}
+                {% endcall %}
+
+                <td>
+                    {% if e.hospital1 == user.hospital.id %}
+                        Giving {{ e.count }} of <a href="#" data-toggle="tooltip" title="{{ e.ppe_ref.desc }}">{{ e.ppe_ref.sku }}</a> to {{ e.hospital2_ref.name }}
```

Comment:
These `<a>` tags are used by Bootstrap to generate a nice little popup when you hover over the SKU with the full description of the PPE.